src/weapon/w_029.c

```diff
@@ -47,7 +49,151 @@ void func_ptr_80170014(Entity* self) {}
 
 int GetWeaponId(void) { return 29; }
 
-INCLUDE_ASM("weapon/nonmatchings/w_029", EntityWeaponShieldSpell);
+void EntityWeaponShieldSpell(Entity* self) {
+    Primitive* prim;
+    s32 var_v0;
+    u16 selfX;
+    u16 selfY;
+
+    switch (self->step) {
+    case 0:
+        self->primIndex = g_api.AllocPrimitives(PRIM_GT4, 1);
+        if (self->primIndex == -1) {
+            DestroyEntity(self);
+            return;
+        }
+        SetSpriteBank1(g_Animset);
+        if (g_HandId != 0) {
+            self->animSet = -0x7FEE;
+            self->ext.weapon.childPalette = 0x128;
```

Comment:
I think it probably makes sense to use the weapon29 extension for everything in this function now that it exists

Reply:
For now I'd like to keep weapon29 simplified. The biggest thing is that its only mismatch is right at the beginning of the struct. As we identify the other members of the overall weapon struct, it would be nice to not have to copy each one into weapon29. 7C/7D is the only oddball member of the struct so I think we can use the broader `weapon` struct to refer to the places where weapon29 behaves the same as a normal weapon. This is why I only defined the first two members for weapon29 in entity.h.

Reply:
@Xeeynamo What's your preference here?

Reply:
When I asked prior to updating this PR here: https://discord.com/channels/1079389589950705684/1087866009983139870/1156579697086439464 xeeynamo expressed no preference. Not trying to argue my case or anything, but if it's helpful for you, I think he already answered.